### PR TITLE
Unread status for channels

### PR DESCRIPTION
**Pull request by skddc**
This adds a visibility status to channels, and marks them as unread when new messages arrive, while they're not visible.

Connected to #46

### CODE REVIEW

**Comment by skddc** [comment]
Sorry for the typo in the branch name. ;)

File: app/models/channel.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ export default Ember.Object.extend({
   connected: false,
   sockethubChannelId: null,
   topic: null,
+  unreadMessages: false,
+  unreadMentions: false,
+  visible: false, // Current/active channel
```
Comment (galfert) [inline app/models/channel.js]: Not sure `visible` is a good choice. Shouldn't it rather be `active`?

`visible` makes me think you could also hide channels somehow.

**Comment by galfert** [comment]
Works good.

I added the missing code for marking the channel when receiving unread mentions.

**Comment by skddc** [comment]
Nice!

File: app/models/channel.js (hunk 1 above)
Comment (skddc) [inline app/models/channel.js]: I thought about it and at the time decided that "visible" was more apt. Thinking about it again, it's all the same to me. I just didn't like how "active" made it sound like other channels were somehow inactive, when this very feature is about indicating channel activity in channels that are not currently open/visible.

**Comment by silverbucket** [comment]
What about notify / notification?

On Sat, Jan 14, 2017, 13:33 Sebastian Kippe <notifications@github.com>
wrote:

*@skddc* commented on this pull request.
------------------------------

In app/models/channel.js <https://github.com/67P/hyperchannel/pull/56>:

> @@ -8,6 +8,9 @@ export default Ember.Object.extend({
   connected: false,
   sockethubChannelId: null,
   topic: null,
+  unreadMessages: false,
+  unreadMentions: false,
+  visible: false, // Current/active channel

I thought about it and at the time decided that "visible" was more apt.
Thinking about it again, it's all the same to me. I just didn't like how
"active" made it sound like other channels were somehow inactive, when this
very feature is about indicating channel activity in channels that are not
currently open/visible.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/67P/hyperchannel/pull/56>, or mute the thread

**Comment by skddc** [comment]
`notify` could be a computed property, determining if notifications should be shown based on both visibility of the channel as well as per-channel notification settings. `visible` is only used to determine if the channel is open/visible/active/current or not.

(By the way, your whole email was copied to the GitHub comment. You might want to check your settings for reply format.)

**Comment by silverbucket** [comment]
Oh yeah I forgot github reply from email does that.

Maybe `updated` flag? to indicate that there have been updates to the
channel since you last looked?

**Comment by skddc** [comment]
This is all implemented already, and it's using `unreadMessages` and `unreadMentions` as properties on the channel model for that.

**Comment by skddc** [comment]
Added some tests for the unread class function, but we'll have to look into integration tests for what we have as well as future features asap.

**Pull request merged**